### PR TITLE
[MBL-2062] Cleanup NoShippingAtCheckout Feature Flag

### PR DESCRIPTION
**Pull request by scottkicks**
# 📲 What

1st step in cleaning up legacy checkout code. Code that isn't prefaced with NoShipping)

Because we're defaulting behavior to this No Shipping flow, and removing the legacy classes in my next PR, you'll find that the legacy view controller snapshots have changes. This is because they are reacting to the No Shipping flow behavior. Since, those will be deleted in my next PR it's no big deal, but it does make the # of file changes a lot here.

# 🤔 Why

Part of our effort to cleanup our checkout code and remove the now legacy checkout classes.

# 🛠 How

Deletes `featureNoShippingEnabled()` from `RemoteConfigFeature` and cleans up all uses of it.

# ✅ Acceptance criteria

- [x] No Shipping At Checkout is now the only version of checkout
- [x] No Shipping Flow experiences no change functionally or visually
- [x] No Shipping Feature Flag has been removed from Remote Config Feature Flags menu
- [x] There's no way to enter the non "no shipping" flow
- [x] All tests pass

### CODE REVIEW

**Comment by nativeksr** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Library/SharedFunctions.swift [hunk 1]
```diff
@@ -367,7 +367,7 @@ public func rewardsCarouselCanNavigateToReward(_ reward: Reward, in project: Pro
 
   let isBacking = userIsBacking(reward: reward, inProject: project)
   let isAvailableForNewBacker = rewardIsAvailable(reward) && !isBacking
-  let isAvailableForExistingBackerToEdit = (isBacking && (reward.hasAddOns || featureNoShippingAtCheckout()))
+  let isAvailableForExistingBackerToEdit = isBacking
```
Comment (scottkicks) [inline Library/SharedFunctions.swift:370]: - In the new “no shipping” flow, that is now the standard, backers should be able to edit rewards regardless of if that reward hasAddOns. This is because backers should have access to change the bonus or pledge amount.

File: Library/ViewModels/ManagePledgeViewModel.swift
Comment (scottkicks) [inline Library/ViewModels/ManagePledgeViewModel.swift:1]: updatePledge is not available in the no shipping flow, so we can remove it completely now

File: Kickstarter-iOS/Features/PledgePaymentMethods/Controller/PledgeRewardsSummaryTotalViewControllerTests.swift [hunk 2]
```diff
@@ -50,17 +50,12 @@ final class PledgeRewardsSummaryTotalViewControllerTests: TestCase {
       }
```
Comment (amy-at-kickstarter) [inline Kickstarter-iOS/Features/PledgePaymentMethods/Controller/PledgeRewardsSummaryTotalViewControllerTests.swift:50]: Nice, good to see these tests cleaned up. Did any other tests change or break now that the feature flag is permanently on?

File: Library/ViewModels/RewardAddOnSelectionContinueCTAViewModelTests.swift [hunk 3]
```diff
@@ -54,28 +54,28 @@ final class RewardAddOnSelectionContinueCTAViewModelTests: TestCase {
...
-    self.vm.inputs.configure(with: (1, true, false, nil))
-
-    self.buttonTitle.assertValues([
-      "Skip add-ons",
-      "Continue with 2 add-ons",
-      "Skip add-ons",
-      "Continue with 1 add-on"
-    ])
-  }
+//  func testButtonTitle() {
```
Comment (amy-at-kickstarter) [inline Library/ViewModels/RewardAddOnSelectionContinueCTAViewModelTests.swift:57]: Was this intentionally left commented out?

File: Library/ViewModels/RewardCardContainerViewModelTests.swift [hunk 4]
```diff
@@ -81,61 +81,61 @@ final class RewardCardContainerViewModelTests: TestCase {
...
-      "Selected",
-      "Selected",
-      "Selected",
-      "Selected",
-      "Selected",
-      "Selected",
-      "Selected"
-    ])
-  }
+//  func testLive_BackedProject_BackedReward() {
```
Comment (amy-at-kickstarter) [inline Library/ViewModels/RewardCardContainerViewModelTests.swift:84]: Was this intentionally left commented out?

**Review by amy-at-kickstarter — APPROVED** [review]
WOW, that's a lotta deleted code! This LGTM. I've got two non-blocking comments:

First: are you confident that the critical paths for backing and updating in the no-shipping pathway are covered by the remaining tests? It's a little hard to tell just from the deletes, but now that this is the "real" pledge flow it would be good to double-check. Anywhere we were testing `update` should also have tests for `updateReward` and `changePaymentMethod`, for example.

Second: Wow, this ended up being a bigger change than I expected! Too late now, but in the future, I'd submit a change like this view-controller-by-view-controller. My guess is that you started by deleting the flag and figuring out which files needed to change, right? A nice way around that would be to split up your changes by commit like
```
commit1 Remove feature flag
commit2 Delete feature flag from FooViewModel
commit3 Delete feature flag from BarViewModel
```
Then you could cherry-pick `commit2` and `commit3` into their own individual branches, and they can be reviewed separately. Then deleting the feature flag (`commit1`) could be submitted for review last, and would just be a one or two line change.

Thanks for fighting the good fight, looking forward to seeing this go out 🫡

File: Library/ViewModels/ManagePledgeViewModelTests.swift [hunk 5]
```diff
@@ -679,103 +670,6 @@ internal final class ManagePledgeViewModelTests: TestCase {
     }
   }
 
-  func testGoToUpdatePledge() {
```
[the remark this answers is not in the document]
Comment (scottkicks) [reply Library/ViewModels/ManagePledgeViewModelTests.swift:682]: Yea! There are existing tests that cover these outputs...
 Screenshot 2025-02-03 at 10 45 02 AM

File: Library/ViewModels/RewardAddOnSelectionContinueCTAViewModelTests.swift (hunk 3 above)
Comment (scottkicks) [reply Library/ViewModels/RewardAddOnSelectionContinueCTAViewModelTests.swift:57]: ah yea forgot to push my last commit. fixing this now!

File: Library/ViewModels/RewardCardContainerViewModelTests.swift (hunk 4 above)
Comment (scottkicks) [reply Library/ViewModels/RewardCardContainerViewModelTests.swift:84]: ah yea forgot to push my last commit. fixing this now!

File: Library/ViewModels/ManagePledgeViewModelTests.swift (hunk 5 above)
[the remark this answers is not in the document]
Comment (scottkicks) [reply Library/ViewModels/ManagePledgeViewModelTests.swift:682]: When it comes to updating rewards/pledge and bonus amounts and changing payment methods, the No Shipping feature work didn't change much except removing the "Update Pledge" menu option. We did that because the pledge/bonus amount selectors were removed from the checkout screen, rendering that menu option unnecessary. 

Backed vs. unbacked reward card availability and the rest of the "Edit Reward" and "Change Payment" cases are still covered. I've also spent a decent amount of time testing various edge cases with the changes in this PR.

[the remark this answers is not in the document]
Comment (amy-at-kickstarter) [reply Library/ViewModels/ManagePledgeViewModelTests.swift:682]: Awesome!

File: Library/ViewModels/RewardCardContainerViewModelTests.swift [hunk 6]
```diff
@@ -81,61 +81,61 @@ final class RewardCardContainerViewModelTests: TestCase {
...
+      "Continue",
+      "Continue",
+      "Continue",
+      "Continue",
+      "Continue",
+      "Continue",
+      "Continue",
+      "Continue"
+    ])
+  }
```
Comment (amy-at-kickstarter) [inline Library/ViewModels/RewardCardContainerViewModelTests.swift:138]: 👍

File: Kickstarter-iOS/Features/PledgePaymentMethods/Controller/__Snapshots__/PledgeRewardsSummaryTotalViewControllerTests/testViewWithPledgeInFull.lang_en_device_pad.png
Comment (ifosli) [inline Kickstarter-iOS/Features/PledgePaymentMethods/Controller/__Snapshots__/PledgeRewardsSummaryTotalViewControllerTests/testViewWithPledgeInFull.lang_en_device_pad.png:1]: Nit: These snapshots are only new because the test names changed. It would be nice to mark them as moved instead of added (I don't think that's worth doing now but definitely keep it in mind when you rename files!). Do please delete the old snapshots, though!

File: Kickstarter-iOS/Features/PledgeView/Controllers/__Snapshots__/PledgeViewControllerTests/testView_HasAddOns_ShippingSelected.lang_de_device_pad_LoggedIn.png
Comment (ifosli) [inline Kickstarter-iOS/Features/PledgeView/Controllers/__Snapshots__/PledgeViewControllerTests/testView_HasAddOns_ShippingSelected.lang_de_device_pad_LoggedIn.png:1]: Looks like the shipping location got deleted again... That's not ideal but I don't think it's worth fixing if we're deleting the entire class anyways.

File: Kickstarter-iOS/Features/ProjectPage/Controller/ProjectPageViewController.swift [hunk 7]
```diff
@@ -638,14 +638,9 @@ public final class ProjectPageViewController: UIViewController, MessageBannerVie
...
   private func goToRewards(project: Project, refTag: RefTag?) {
-    if featureNoShippingAtCheckout() {
-      /// Render rewards carousel that has the shipping location dropdown
-      let vc = WithShippingRewardsCollectionViewController.controller(with: project, refTag: refTag)
-      self.present(vc, animated: true)
-    } else {
-      let vc = RewardsCollectionViewController.controller(with: project, refTag: refTag)
-      self.present(vc, animated: true)
-    }
+    /// Render rewards carousel that has the shipping location dropdown
```
Comment (ifosli) [inline Kickstarter-iOS/Features/ProjectPage/Controller/ProjectPageViewController.swift:641]: Same here; this comment doesn't seem helpful anymore

File: Kickstarter-iOS/Features/RewardAddOnSelection/Controller/__Snapshots__/RewardAddOnSelectionViewControllerTests/testView_EmptyState.lang_de_device_phone4inch.png
Comment (ifosli) [inline Kickstarter-iOS/Features/RewardAddOnSelection/Controller/__Snapshots__/RewardAddOnSelectionViewControllerTests/testView_EmptyState.lang_de_device_phone4inch.png:1]: This is another example of tests that would've been easier to review if they'd been deleted immediately, instead of being changed now and deleted later.

File: Kickstarter-iOS/Features/RewardAddOnSelection/Views/RewardAddOnCardView.swift [hunk 8]
```diff
@@ -96,9 +96,7 @@ public final class RewardAddOnCardView: UIView {
       self.rewardLocationStackView
     ]
 
-    if featureNoShippingAtCheckout() {
-      stackViews.insert(self.estimatedShippingStackView, at: 3)
-    }
+    stackViews.insert(self.estimatedShippingStackView, at: 3)
```
Comment (ifosli) [inline Kickstarter-iOS/Features/RewardAddOnSelection/Views/RewardAddOnCardView.swift:99]: nit: I think it'd be cleaner now to add this to the list above instead of inserting it after the list is created.

File: Kickstarter-iOS/Features/RewardAddOnSelection/Views/RewardAddOnCardView.swift [hunk 9]
```diff
@@ -270,9 +268,7 @@ public final class RewardAddOnCardView: UIView {
       self.stepperStackView
     ]
 
-    if featureNoShippingAtCheckout() {
-      rootSubviews.insert(self.estimatedShippingStackView, at: 4)
-    }
+    rootSubviews.insert(self.estimatedShippingStackView, at: 4)
```
Comment (ifosli) [inline Kickstarter-iOS/Features/RewardAddOnSelection/Views/RewardAddOnCardView.swift:271]: Same here; no need to insert after list has been created.

File: Kickstarter-iOS/Features/RewardsCollection/Views/RewardCardView.swift [hunk 10]
```diff
@@ -75,9 +75,7 @@ public final class RewardCardView: UIView {
       self.rewardLocationStackView
     ]
 
-    if featureNoShippingAtCheckout() {
-      stackViews.insert(self.estimatedShippingStackView, at: 4)
-    }
+    stackViews.insert(self.estimatedShippingStackView, at: 4)
```
Comment (ifosli) [inline Kickstarter-iOS/Features/RewardsCollection/Views/RewardCardView.swift:78]: Same here; no need to insert after list has been created.

File: Kickstarter-iOS/Features/RewardsCollection/Views/RewardCardView.swift [hunk 11]
```diff
@@ -259,9 +257,7 @@ public final class RewardCardView: UIView {
       self.pillsView
     ]
 
-    if featureNoShippingAtCheckout() {
-      baseSubviews.insert(self.estimatedShippingStackView, at: 4)
-    }
+    baseSubviews.insert(self.estimatedShippingStackView, at: 4)
```
Comment (ifosli) [inline Kickstarter-iOS/Features/RewardsCollection/Views/RewardCardView.swift:260]: Same here; this could've been added to the initial list

File: Library/ViewModels/PledgeShippingLocationViewModel.swift [hunk 12]
```diff
@@ -54,18 +54,11 @@ public final class PledgeShippingLocationViewModel: PledgeShippingLocationViewMo
       .map { $0.3 }
 
     let shippingShouldBeginLoading = reward
-      .map { featureNoShippingAtCheckout() ? true : $0.shipping.enabled }
+      .mapConst(true)
```
Comment (ifosli) [inline Library/ViewModels/PledgeShippingLocationViewModel.swift:57]: Nit: Can we get rid of this signal entirely now? Probably in a separate pr?

File: Library/ViewModels/RewardCardContainerViewModelTests.swift [hunk 13]
```diff
@@ -115,25 +115,25 @@ final class RewardCardContainerViewModelTests: TestCase {
     self.pledgeButtonTitleText.assertValueCount(self.allRewards.count)
 
     self.pledgeButtonStyleType.assertValues([
-      .green, .black, .black, .black, .black, .black, .black, .black, .black, .black
+      .green, .green, .green, .green, .green, .green, .green, .green, .green, .green
     ])
     self.pledgeButtonEnabled.assertValues(
-      [true, false, false, false, false, false, false, false, false, false]
+      [true, true, true, true, true, true, true, true, true, true]
```
Comment (ifosli) [inline Library/ViewModels/RewardCardContainerViewModelTests.swift:121]: Nit: All of the `assertValues` in this test are super repetitive; it'd be nice to replace them with `assertLastValue` instead.

File: Library/ViewModels/RewardCardContainerViewModelTests.swift [hunk 14]
```diff
@@ -430,60 +430,60 @@ final class RewardCardContainerViewModelTests: TestCase {
...
-    self.pledgeButtonHidden.assertValues([false, false, false, false, false])
-    self.pledgeButtonTitleText.assertValues([
-      "Selected",
-      "Selected",
-      "Selected",
-      "Selected",
-      "Selected"
-    ])
-  }
+//  func testLive_BackedProject_BackedReward_Errored() {
```
Comment (ifosli) [inline Library/ViewModels/RewardCardContainerViewModelTests.swift:433]: Why is this commented out? I'm guessing this is an accident, but if it's on purpose, please add a comment explaining why/link a ticket to fix/update/delete the test!

**Review by ifosli — COMMENTED** [review]
Overall, very exciting! I left a bunch of nits - up to you what you want to bother dealing with at this point. My main comment for future prs like this - make them smaller, please! If you go class by class (ex hardcode the flag to "true" first instead of deleting it entirely), it'll be easier to just delete tests that we don't need anymore, instead of accidentally modifying them, the way that we see in this pr.
